mono/mini/debugger-agent.c

```diff
@@ -4924,6 +4924,11 @@ debugger_agent_breakpoint_from_context (MonoContext *ctx)
 
 	tls = (DebuggerTlsData *)mono_native_tls_get_value (debugger_tls_id);
 	g_assert (tls);
+	
+	//if a thread was suspended and doesn't have any managed stack, it was considered as terminated, 
+	//but it wasn't really terminated because it can execute managed code again, and stop in a breakpoint so here we set terminated as FALSE
+	tls->terminated = FALSE;
```

Comment:
Do we need `tls->terminated = FALSE` in `debugger_agent_single_step_from_context`, too?

Reply:
I don't think so, because if we are in debugger_agent_single_step_from_context  it is because we have already executed debugger_agent_breakpoint_from_context.